static/js/formatters-internal.js

```diff
@@ -281,134 +281,97 @@ export function joinList(list, separator) {
-      desiredWidth = Number.parseInt(desiredDims[0]);
-      if (Number.isNaN(desiredWidth)) {
-        throw new Error("Invalid width specified");
-      }
+  if (desiredDims[0] !== '') {
+    desiredWidth = Number.parseInt(desiredDims[0]);
+    if (Number.isNaN(desiredWidth)) {
+      throw new Error("Invalid width specified");
     }
+  } else {
+    desiredWidth = atLeastAsLarge ? 1 : Number.parseInt(fullSizeDims[0]);
+  }
 
-    if (desiredDims[1] !== '') {
-      desiredHeight = Number.parseInt(desiredDims[1]);
-      if (Number.isNaN(desiredHeight)) {
-        throw new Error("Invalid height specified");
-      }
+  if (desiredDims[1] !== '') {
+    desiredHeight = Number.parseInt(desiredDims[1]);
+    if (Number.isNaN(desiredHeight)) {
+      throw new Error("Invalid height specified");
     }
-    const thumbnails = image.thumbnails
-      .filter(thumb => thumb.width && thumb.height)
-      .sort((a, b) => b.width - a.width);
-    return atLeastAsLarge
-      ? _getSmallestThumbnailOverThreshold(thumbnails, desiredWidth, desiredHeight)
-      : _getLargestThumbnailUnderThreshold(thumbnails, desiredWidth, desiredHeight);
+  } else {
+    desiredHeight = atLeastAsLarge ? 1 : Number.parseInt(fullSizeDims[1]);
   }
 
-  const result = imageBySizeEntity(img, size, atLeastAsLarge);
+  const urlWithDesiredDims = urlBeforeDimensions + desiredWidth + 'x' + desiredHeight + extension;
+
+  const dynamicUrl = atLeastAsLarge
+    ? _replaceUrlHost(urlWithDesiredDims, 'dynl.mktgcdn.com')
+    : _replaceUrlHost(urlWithDesiredDims, 'dynm.mktgcdn.com');
 
   return Object.assign(
     {},
-    img,
+    image,
     {
-      url: result.replace('http://', 'https://')
+      url: dynamicUrl.replace('http://', 'https://')
     }
   );
 }
 
 /**
- * Gets the smallest thumbnail that is over the min width and min height.
- * If no thumbnails are over the given thresholds, will return the closest one.
- *
- * This method assumes all thumbnails have the same aspect ratio, and that
- * thumbnails are sorted in descending size.
- *
- * @param {Array<{{url: string, width: number, height: number}}>} thumbnails 
- * @param {number|undefined} minWidth 
- * @param {number|undefined} minHeight 
- * @returns {string}
+ * Splits a url into two parts at the specified index.
+ * 
+ * @param {string} url The url to be split
+ * @param {number} index The index at which to split the url
+ * @returns {Array<string>} The two parts of the url after splitting
  */
-function _getSmallestThumbnailOverThreshold(thumbnails, minWidth, minHeight) {
-  let index = thumbnails.length - 1;
-  while (index > 0) {
-    const thumb = thumbnails[index];
-    const widthOverThreshold = minWidth ? thumb.width >= minWidth : true;
-    const heightOverThreshold = minHeight ? thumb.height >= minHeight : true;
-    if (widthOverThreshold && heightOverThreshold) {
-      return thumb.url
-    }
-    index--;
-  }
-  return thumbnails[0].url;
+function _splitUrlOnIndex(url, index) {
```

Comment:
probably just splitStringOnIndex? since it's not specific to url

Reply:
changed!